core/threaded/scheduler_LET.c

```diff
@@ -515,5 +609,42 @@ void lf_sched_reaction_epilogue(reaction_t * reaction, int worker_number) {
             lf_decrement_global_tag_barrier_locked();
         }
         lf_mutex_unlock(&mutex);
+
+        // unlock any containing reactor which is modal to enable mode changes in parents
+        #ifdef MODAL_REACTORS
+            _lf_sched_unlock_modal_parents(self, worker_number);
+        #endif
     }
-}
\ No newline at end of file
+}
+
+#ifdef MODAL_REACTORS
+/**
+ * @brief This function should be invoked by the worker thread about to advance time BEFORE it acquires the global mutex.
+ *  It is only relevant if we have modal reactors. It gathers all the Reactors which are about to perform mode changes, and reactors contained within them,
+ *  and acquires there local mutex to make sure there are no LET reactions currently executing in any of them.
+ *  In the case that there are LET reactions, this will block the advancement of time until the LET reaction completes.
+ * 
```

Comment:
```suggestion
 * @brief Acquire the local mutex for any reactor that has pending mode changes.
 * This function should be invoked by the worker thread about to advance time BEFORE it acquires the global mutex.
 *  It is only relevant if we have modal reactors. 
 * It gathers all the Reactors which are about to perform mode changes, and reactors contained within them,
 *  and acquires their local mutex to make sure there are no LET reactions currently executing in any of them.
 *  In the case that there are LET reactions, this will block the advancement of time until the LET reaction completes.
```